aggregator/programs/build.ts

```diff
@@ -5,103 +5,164 @@ import { dirname, parseArgs } from "../deps.ts";
-} catch (error) {
-  if (error.name === "NotFound") {
-    // We don't care that remove failed due to NotFound (why do we need to catch
-    // an exception to handle this normal use case? 🤔)
-  } else {
-    throw error;
+await ensureFreshBuildDir();
+await buildEnvironment();
+await copyTypescriptFiles();
+await buildDockerImage();
+
+console.log("Aggregator build complete");
+
+async function allFiles() {
+  return [
+    ...await shell.Lines("git", "ls-files"),
+    ...await shell.Lines(
+      "git",
+      "ls-files",
+      "--others",
+      "--exclude-standard",
+    ),
+  ];
+}
+
+async function shortContentHash(filePath: string) {
+  const contentHash = (await shell.Line("shasum", "-a", "256", filePath));
+
+  return contentHash.slice(0, 7);
+}
+
+async function BuildName() {
+  const commitShort = (await shell.Line("git", "rev-parse", "HEAD")).slice(
+    0,
+    7,
+  );
+
+  const isDirty =
+    (await shell.Lines("git", "status", "--porcelain")).length > 0;
+
+  const envHashShort = await shortContentHash(`${buildDir}/.env`);
+
+  return [
+    "git",
+    commitShort,
+    ...(isDirty ? ["dirty"] : []),
+    "env",
+    envName,
+    envHashShort,
+  ].join("-");
+}
+
+async function ensureFreshBuildDir() {
+  try {
+    await Deno.remove(buildDir, { recursive: true });
+  } catch (error) {
+    if (error.name === "NotFound") {
+      // We don't care that remove failed due to NotFound (why do we need to
+      // catch an exception to handle this normal use case? 🤔)
+    } else {
+      throw error;
+    }
   }
+
+  await Deno.mkdir(buildDir);
 }
 
-await Deno.mkdir(buildDir);
+async function buildEnvironment() {
+  const repoDotEnv = await Deno.readTextFile(dotEnvPath);
+
+  let networkConfigPaths: { repo: string; build: string } | nil = nil;
+  const buildDotEnvLines: string[] = [];
 
-await Deno.copyFile(dotEnvPath, `${buildDir}/.env`);
+  for (const line of repoDotEnv.split("\n")) {
+    let buildLine = line;
 
-for (const f of await allFiles()) {
-  if (!f.endsWith(".ts")) {
-    continue;
+    if (line.startsWith("NETWORK_CONFIG_PATH=")) {
+      const repoNetworkConfigPath = line.slice(
+        "NETWORK_CONFIG_PATH=".length,
+      );
+
+      const networkConfigHash = await shortContentHash(repoNetworkConfigPath);
+
+      networkConfigPaths = {
+        repo: repoNetworkConfigPath,
+        build: `networkConfig-${networkConfigHash}.json`,
+      };
+
+      // Need to replace this value with a build location because otherwise
+      // this file might not be included in the docker image
+      buildLine = `NETWORK_CONFIG_PATH=${networkConfigPaths.build}`;
+    }
+
+    buildDotEnvLines.push(buildLine);
   }
 
-  console.log("Processing", f);
-  await Deno.mkdir(dirname(`${buildDir}/ts/${f}`), { recursive: true });
-  await Deno.copyFile(f, `${buildDir}/ts/${f}`);
+  if (networkConfigPaths !== nil) {
+    await Deno.copyFile(
+      networkConfigPaths.repo,
+      `${buildDir}/${networkConfigPaths.build}`,
+    );
+  }
```

Comment:
Generally when running containerized services, the container image does not have configuration included in the image (asides from safe/needed defaults) and instead the user provides that via [a variety of options](https://docs.docker.com/engine/reference/run/#overriding-dockerfile-image-defaults), most common being command overrides, env vars, and file/volume mounts. `geth` for example uses command overrides and volume mounts for loading and persisting data https://github.com/jzaki/bls-wallet/blob/main/docker-compose.yml#L26-L34 . We can consider updating how we want to do that in the future.